firebase-test/src/supports-rules.test.ts

```diff
@@ -0,0 +1,175 @@
+      await expectReadToFail(db, kSupportDocPath);
+    });
+
+    it("unauthenticated users can't write multi-class supports", async () => {
+      db = initFirestore();
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc());
+    });
+
+    it("authenticated generic users can't read multi-class supports", async () => {
+      db = initFirestore(genericAuth);
+      await expectReadToFail(db, kSupportDocPath);
+    });
+
+    it("authenticated generic users can't write multi-class supports", async () => {
+      db = initFirestore(genericAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc());
+    });
+
+    it("authenticated teachers can't create multi-class supports without valid uid", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc({ remove: ["uid"] }));
+    });
+
+    it("authenticated teachers can't create multi-class supports without valid classes", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc({ remove: ["classes"] }));
+    });
+
+    it("authenticated teachers can't create multi-class supports without valid context_id", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc({ remove: ["context_id"] }));
+    });
+
+    it("authenticated teachers can't create multi-class supports without valid type", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc({ remove: ["type"] }));
+    });
+
+    it("authenticated teachers can't create multi-class supports without content", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc({ remove: ["content"] }));
+    });
+
+    it("authenticated teachers can create valid multi-class supports", async () => {
+      db = initFirestore(teacherAuth);
+      await expectWriteToSucceed(db, kSupportDocPath, specSupportDoc());
+    });
+
+    it("authenticated teacher users can't create multi-class supports for other teachers", async () => {
+      db = initFirestore(teacher2Auth);
+      await expectWriteToFail(db, kSupportDocPath, specSupportDoc());
+    });
+
+    it("authenticated teachers can read multi-class supports for their class", async () => {
+      db = initFirestore(teacherAuth);
+      await adminWriteDoc(kSupportDocPath, specSupportDoc())
+      await expectReadToSucceed(db, kSupportDocPath);
+    });
+
+    it("authenticated teachers can read multi-class supports for their co-teachers' classes", async () => {
+      db = initFirestore(teacher2Auth);
+      await adminWriteDoc(kSupportDocPath, specSupportDoc({ add: { classes: [thisClass, otherClass] } }));
+      await expectReadToSucceed(db, kSupportDocPath);
+    });
+
+    it("authenticated teachers can update their own multi-class supports", async () => {
```

Comment:
Maybe you want to add additional tests showing how teachers can create or update supports that have multiple classes as long as one class is their own class.

Reply:
As mentioned above, I should add those tests, although the access should fail due to the `resourceMatchUserId()` test.